text/0000-simple_unwind_annotation.md

```diff
@@ -0,0 +1,75 @@
+- Feature Name: `simple_unwind_attribute`
+- Start Date: 2019-08-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+Provides an annotation to permit functions with explicit ABI specifications
+(such as `extern "C"`) to unwind
+
+# Motivation
+[motivation]: #motivation
+
+TODO
+
+- soundness & optimization
+- libjpeg/mozjpeg
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+TODO
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+TODO: incorporate changes from discussion here:
+https://github.com/rust-lang/rust/pull/63909
+
+By default, Rust assumes that an external function imported with extern "C"
+cannot unwind, and Rust will abort if a panic would propagate out of a Rust
+function exported with extern "C" function. If you specify the #[unwind]
+attribute on an extern "C" function, Rust will instead allow an unwind (such as
```

Comment:
```suggestion
attribute on a function with a non-"Rust" ABI, Rust will instead allow an unwind (such as
```

To be clarified: the annotation can be applied to _either_ function definitions (`extern "C" fn...`) or function declarations-without-definitions (`extern "C" { fn ...`; I'm not sure if there's a better term for these declarations).

Without this annotation, what is the behavior supposed to be? I'm pretty sure that for maximum soundness, definitions should always abort and declarations should be assumed _not_ to abort. Should declarations without `#[unwind(allowed)]` be given automatic "wrapper" functions that would catch unwinding and trigger an `abort`, or should they simply propagate the exception as normal?